### PR TITLE
Bugfix: LEAF-2473 - New Row logic update

### PR DESCRIPTION
**Pull request by aerinkayne**
Addresses the situation of the New Row button not appearing if no rows are returned from the search.

### CODE REVIEW

**Review by AmandaSumner — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]
Looks good

**Review by raphaelfretter — APPROVED** [review]
Looks good

File: LEAF_Request_Portal/templates/view_reports.tpl [hunk 1]
```diff
@@ -888,6 +888,37 @@ var indicatorSort = {}; // object = indicatorID : sortID
...
 let gridColorData = {}; //object updated with id: color
 let tempColorData = {}; //object updated with id: color
+let isOneFormType = false;
+
+/**
+ * Purpose: Check if only one type of form could logically be returned and,
+ * if so, update global variables isOneFormType (bool) and categoryID (string).
+ * @param searchQueryTerms - variable with result of leafSearch.getLeafFormQuery().getQuery().terms
+ */
+function checkIfOneTypeSearchedAndUpdate(searchQueryTerms) {
```
Comment (Pelentan) [inline LEAF_Request_Portal/templates/view_reports.tpl:898]: Need some type checking on searchQueryTerms.  I know IE doesn't like it in the call, but inside the function we need to ensure it's actually and object so later calls don't fail badly.

**Review by Pelentan — CHANGES_REQUESTED** [review]

**Review by pete-nerantzinis — APPROVED** [review]
Looks good

**Review by Pelentan — APPROVED** [review]

**Review by beingforthebenefit — APPROVED** [review]

**Pull request merged**